src/main/TreeQuery.cs

```diff
@@ -0,0 +1,209 @@
+                        if (result.AvatarUrl != null && result.AvatarUrl.Any())
+                        {
+                            ProcessAvatarUrlParameters(result, array, typeFromHandle);
+                        }
+
+                        result2 = true;
+                    }
+                    catch (ArgumentException)
+                    {
+                        // Handle invalid enum values gracefully
+                        result = null;
+                        result2 = false;
+                    }
+                }
+            }
+
+            return result2;
+        }
+
+        private static Dictionary<string, List<string>> ParseQueryString(string queryString)
+        {
+            var parameters = new Dictionary<string, List<string>>(StringComparer.OrdinalIgnoreCase);
+            string[] pairs = queryString.Split('&');
+
+            foreach (string pair in pairs)
+            {
+                if (string.IsNullOrEmpty(pair)) continue;
+
+                int equalIndex = pair.IndexOf('=');
+                if (equalIndex > 0)
+                {
+                    string key = HttpUtility.UrlDecode(pair.Substring(0, equalIndex));
+                    string val = HttpUtility.UrlDecode(pair.Substring(equalIndex + 1));
+
+                    if (!parameters.ContainsKey(key))
+                        parameters[key] = new List<string>();
+                    parameters[key].Add(val);
+                }
+                else if (equalIndex == 0)
+                {
+                    // Handle case where key is empty but there's a value
+                    continue;
+                }
+                else
+                {
+                    // Handle case where there's no equals sign (key without value)
+                    string key = HttpUtility.UrlDecode(pair);
+                    if (!parameters.ContainsKey(key))
+                        parameters[key] = new List<string>();
+                    parameters[key].Add("");
+                }
+            }
+
+            return parameters;
+        }
+
+        private static List<string> ConvertParametersToArray(Dictionary<string, List<string>> parameters)
+        {
+            var parameterArray = new List<string>();
+            foreach (var kvp in parameters)
+            {
+                foreach (var val in kvp.Value)
+                {
+                    parameterArray.Add($"{kvp.Key}={val}");
+                }
+            }
+            return parameterArray;
+        }
+
+        private static IEnumerable<string> GetDecodedValues(Dictionary<string, List<string>> parameters, string key)
+        {
+            if (parameters.ContainsKey(key))
+            {
+                // Values are already decoded in ParseQueryString method
+                return parameters[key];
+            }
+            return null;
+        }
+
+        private static void ProcessAvatarUrlParameters(TreeQuery result, string[] array, Type typeFromHandle)
```

Comment:
No need to parse the avatarUrl. This was not specified in the acceptance criteria.